design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,357 @@
+      },
+      inputs:       %i(provider_uri token_jwt)
+    ) do
+
+      def call
+		certs = @fetch_provider_certificate.(
+		  provider_uri: @provider_uri
+		)     
+
+		# indicate that we should verify the issuer and audience claims
+		options = { iss: provider_uri, verify_iss: true, verify_aud: true }
+
+        @decoded_token = JWT.decode(
+          @token_jwt,
+          @certs, # JWKs retrieved from the 
+          true, # tells the decoder to verify the token
+          options
+        )
+      end
+    end
+  end
+end
+```
+ 
+#### validate_application_identity
+
+The validate_application_identity function will validate if the Azure VM can authenticate with Conjur and fetch secrets based on information extracted from the provided Azure token. 
+This function will parse the `xms_mirid` field of the Azure token provided by the Azure VM, extracting resource group, system/user assigned identity, and client ID. Once extracted, 
+we will compare them to the annotations of the Conjur host for this Azure resource. If they match, the next validation test will run. If all validations run without raising an error, 
+the Azure VM resource will receive a Conjur access token in return. All further requests that this Azure VM sends will need to have this Conjur access token in the header.
+
+##### Host annotation / xms_mirid mapping
+
+Recap, Conjur Host annotations that need to be filled and match based on "xms_mirid" claim in the Azure token:
+
+```yaml
+  authn-azure/subscription-id: test-subscription 
+  authn-azure/resource-group: test-group 
+  authn-azure/user-assigned-identity: test-app-pipeline 
+OR 
+  authn-azure/system-assigned-identity: ie_qWCXhXxt1zIEsu4c7acQVGn4
+```
+
+An example of a "xms_mirid" claim is listed below. We will use what is defined in this claim and compare it against the Conjur identity for the Azure resource. If they match, then we can confirm that the 
+Azure resource that has made the request has a Conjur identity.
+
+```yaml
+"xms_mirid": "/subscriptions/<subscription_id>/resourcegroups/<resource_group>/providers/Microsoft.ContainerInstance/containerGroups/<assigned_identity>"
+```
+
+| Host annotation                                                      | xms_mirid              |
+|----------------------------------------------------------------------|------------------------|
+| `authn-azure/subscription-id`                                        | subscription_id        |
+| `authn-azure/resource-group`                                         | resource_group         |
+| `authn-azure/user-assigned-identity` or `system-assigned-identity`   | assigned_identity      |
+
+### Backwards compatibility
+
+None because this is a new feature.
+
+### Performance
+
+Azure authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+
+- Conjur
+
+- DAP
+
+## Test Plan
+
+TBT
+
+### Integration tests
+
+#### Functionality
```

Comment:
1. Need to verify more 4xx scenarios such as:
- 400 - size to large , invalid chars
- 408 - request timeout if we support it. 
- 429 - too many requests 
2. VM sends authentication request with a **invalid** Azure token